### PR TITLE
test: update unit tests to use temporary folders instead of writing files to the main codebase folders

### PR DESCRIPTION
**Pull request by MiroslavGatsanoga**
**Description**:

Fixes for most of the temp test directories from `.gitignore`; the only one left (`**/swirlds-tmp`) will be addressed in separate PR.

**Related issue(s)**:

https://github.com/hashgraph/hedera-services/issues/6714

**Notes for reviewer**:

In order to verify that the previously used test files and directories are not being used anymore run the following:
```
./gradlew -x :test-clients:test test timingSensitive  --rerun-tasks
```

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/cli-clients/src/test/java/com.hedera.services.cli.sign.test/RecordStreamSigningUtilsTest.java
Comment (jsync-swirlds) [inline hedera-node/cli-clients/src/test/java/com.hedera.services.cli.sign.test/RecordStreamSigningUtilsTest.java:1]: Is it just me, or is the folder name off for this file?

Comment (MiroslavGatsanoga) [reply hedera-node/cli-clients/src/test/java/com.hedera.services.cli.sign.test/RecordStreamSigningUtilsTest.java:1]: Hmm that's strange, it is that way in develop: https://github.com/hashgraph/hedera-services/tree/develop/hedera-node/cli-clients/src/test/java/com.hedera.services.cli.sign.test

Just to be clear, by off you mean that `sign` and `test` should be swapped i.e. we should instead have `hedera-node/cli-clients/src/test/java/com.hedera.services.cli.test.sign`?

Comment (MiroslavGatsanoga) [reply hedera-node/cli-clients/src/test/java/com.hedera.services.cli.sign.test/RecordStreamSigningUtilsTest.java:1]: Pushed a commit with the renaming.

**Review by jsync-swirlds — APPROVED** [review]

**Review by derektriley — APPROVED** [review]

File: platform-sdk/platform-apps/tests/PlatformTestingTool/src/test/java/com/swirlds/demo/merkle/map/SaveExpectedMapHandlerTest.java [hunk 1]
```diff
@@ -106,12 +101,12 @@ private byte[] generateRandomContent() {
     // Serializes and deserializes the expected map with all valid keys
     @Test
     public void serializeAndDeserializePositiveTest() throws IOException {
-        String jsonValue = serializeThrowing(expectedMap, new File("."), expectedMapName, true);
+        final var jsonValue = serializeThrowing(expectedMap, new File(tmpDir.toString()), expectedMapName, true);
```
Comment (poulok) [inline platform-sdk/platform-apps/tests/PlatformTestingTool/src/test/java/com/swirlds/demo/merkle/map/SaveExpectedMapHandlerTest.java:104]: The platform convention is to not use `var`

File: platform-sdk/platform-apps/tests/PlatformTestingTool/src/test/java/com/swirlds/demo/platform/MapValueSerializableTest.java [hunk 2]
```diff
@@ -373,8 +373,8 @@ private <T extends MapValue & Keyed<MapKey>> void compareTwoFCMs(
     }
 
     @Test
-    void serializeExpectedFCMFamilyTest() {
-        String savePath = "src/test/saved.txt";
+    void serializeExpectedFCMFamilyTest(@TempDir Path tmpDir) {
```
Comment (poulok) [inline platform-sdk/platform-apps/tests/PlatformTestingTool/src/test/java/com/swirlds/demo/platform/MapValueSerializableTest.java:376]:
```suggestion
    void serializeExpectedFCMFamilyTest(final @TempDir Path tmpDir) {
```

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/SignedStateFileReadWriteTest.java [hunk 3]
```diff
@@ -147,7 +147,8 @@ void writeSavedStateToDiskTest() throws IOException {
         final Path addressBookFile = directory.resolve(CURRENT_ADDRESS_BOOK_FILE_NAME);
 
         throwIfFileExists(stateFile, hashInfoFile, settingsUsedFile, directory);
-        final Configuration configuration = changeConfigAndConfigHolder("data/saved");
+        final var configDir = testDirectory.resolve("data/saved").toString();
```
Comment (poulok) [inline platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/SignedStateFileReadWriteTest.java:150]: `var`

Comment (poulok) [inline platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/SignedStateFileReadWriteTest.java:150]:
```suggestion
        final String configDir = testDirectory.resolve("data/saved").toString();
```

File: platform-sdk/platform-apps/tests/PlatformTestingTool/src/test/java/com/swirlds/demo/merkle/map/SaveExpectedMapHandlerTest.java [hunk 4]
```diff
@@ -130,13 +125,13 @@ public void serializeAndDeserializePositiveTest() throws IOException {
     @Test
     public void DeserializeNullEntityTypeTest() throws IOException {
         addInvalidKeysToMap();
-        String jsonValue = serializeThrowing(expectedMap, new File("."), expectedMapName, true);
+        final var jsonValue = serializeThrowing(expectedMap, new File(tmpDir.toString()), expectedMapName, true);
```
Comment (poulok) [inline platform-sdk/platform-apps/tests/PlatformTestingTool/src/test/java/com/swirlds/demo/merkle/map/SaveExpectedMapHandlerTest.java:128]:
```suggestion
        final String jsonValue = serializeThrowing(expectedMap, new File(tmpDir.toString()), expectedMapName, true);
```

File: platform-sdk/platform-apps/tests/PlatformTestingTool/src/test/java/com/swirlds/demo/platform/MapValueSerializableTest.java (hunk 2 above)
Comment (MiroslavGatsanoga) [reply platform-sdk/platform-apps/tests/PlatformTestingTool/src/test/java/com/swirlds/demo/platform/MapValueSerializableTest.java]: Updated.

File: platform-sdk/platform-apps/tests/PlatformTestingTool/src/test/java/com/swirlds/demo/merkle/map/SaveExpectedMapHandlerTest.java (hunk 4 above)
Comment (MiroslavGatsanoga) [reply platform-sdk/platform-apps/tests/PlatformTestingTool/src/test/java/com/swirlds/demo/merkle/map/SaveExpectedMapHandlerTest.java]: Updated.

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/SignedStateFileReadWriteTest.java (hunk 3 above)
Comment (MiroslavGatsanoga) [reply platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/SignedStateFileReadWriteTest.java]: Updated.

**Review by poulok — DISMISSED** [review]
Approved for the hashgraph owned files

**Review by jsync-swirlds — APPROVED** [review]

**Review by poulok — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM, ty @MiroslavGatsanoga !

**Review by dalvizu — APPROVED** [review]

**Pull request merged**